src/state_machine.zig

```diff
@@ -1789,6 +1883,20 @@ pub fn StateMachineType(
             // checks.
             if (self.get_transfer(t.id)) |e| return create_transfer_exists(t, e);
 
+            if (!self.validate_regressed_timestamp(operation, t)) {
+                assert(operation == .import_transfers);
+                return .import_timestamp_must_not_regress;
+            }
+
+            // Imported transfers with past timestamp cannot set a timeout for expiration.
+            if (operation == .import_transfers and t.timeout != 0) {
+                assert(t.flags.pending);
+                return .import_timeout_must_be_zero;
+            }
+
+            assert(t.timestamp > dr_account.timestamp);
+            assert(t.timestamp > cr_account.timestamp);
+
```

Comment:
I think these asserts can go above the `if (operation == .import_transfers and t.timeout != 0) {` condition?